ckanext/canada/templates/public/package/comments.html

```diff
@@ -1,9 +1,27 @@
 <section class="indent-large">
 <div class="span-6 margin-top-medium">
     {%- set pkg = c.pkg_dict -%}
-    {%- set comment_list = h.dataset_comments(pkg.id, h.lang()) -%}
-    <h4><a id="comments"></a>{{ _('Comments') }}</h4>
+    {%- set comment_list = h.dataset_comments(request, c, pkg.id, h.lang()) -%}
+
     {%- if comment_list -%}
+      <div class="row">
+        <div class="col-md-4"><h4 style="display:inline-block;">{{ _('Comments') }}({{ c.page.item_count }} )</h4></div>
+
+        {% block page_pagination_top %}
+          <div class="col-md-4"><small>{{ c.page.pager(q=c.q) }}</small></div>
+         {% endblock %}
+
+        <div class="text-right"><small>
+          <form class="form-inline mrgn-bttm-sm mrgn-tp-md form-group form-horizontal" role="form" method="get" >
+            <select id="field-page-limit" name="pagelimit" class="form-control input-sm" onchange="this.form.submit()">
+              <option value="20" {% if c.pagelimit==20 %} selected="selected"{% endif %}>20</option>
```

Comment:
Loops work like you'd expect in Jinja, so do this instead of each one by hand. Will make it easier if we have to change the options or if we turn it into a setting.
```
{% for value in [20, 50, 100] %}
{% endfor %}
```